lib/fielded_search.rb

```diff
@@ -0,0 +1,63 @@
+class FieldedSearch
+  attr_reader :field_pairs
+
+  def initialize(field_pairs)
+    @field_pairs = field_pairs
+  end
+
+  def present?
+    return false unless field_pairs
+    to_s.present?
+  end
+
+  def value_for(key)
+    if field_pairs && field_pairs.key?(key)
+      field_pairs[key]
+    end
+  end
+
+  def to_s
```

Comment:
defining `present?` and `to_h` and `to_s` feels a bit icky to me because those are methods Ruby gives us. Would it make sense to name these methods differently?